addons/account/models/account_payment.py

```diff
@@ -131,6 +131,16 @@ def _onchange_journal(self):
         res = super(account_register_payments, self)._onchange_journal()
         active_ids = self._context.get('active_ids')
         invoices = self.env['account.invoice'].browse(active_ids)
+        payment_currency = self.currency_id or self.journal_id.currency_id or self.journal_id.company_id.currency_id or invoices and invoices[0].currency_id
```

Comment:
Move the fix to a new function : onchange_currency_id

Reply:
`_onchange_currency_id`